src/test/java/test_user_registration/TestUserExistsInteractor.java

```diff
@@ -0,0 +1,128 @@
+import use_cases.user_registration_use_cases.createMailMan;
+
+public class TestUserExistsInteractor {
+    //Objects used for testing
+    // The below anonymous database is created to test cases where the user does not exist.
+    Database userDontExist = new Database() {
+        @Override
+        public User getUser(String username) {
+            return null;
+        }
+
+        @Override
+        public void createUser(String username, String password, String email, String type) {
+
+        }
+
+        @Override
+        public boolean UserExists(String username, String email) {
+            return false;
+        }
+        @Override
+        public boolean UserExists(String username) {
+            return false;
+        }
+    };
+    //The below anonymous database for testing cases where the user exists
+    Database userExists = new Database() {
+        @Override
+        public User getUser(String username) {
+            return null;
+        }
+
+        @Override
+        public void createUser(String username, String password, String email, String type) {
+
+        }
+
+        @Override
+        public boolean UserExists(String username, String email) {
+            return true;
+        }
+
+        @Override
+        public boolean UserExists(String username) {
+            return true;
+        }
+    };
+    //This output boundary used in test cases
+    private class TestOutputBoundary implements UserExistsOutputView {
+        public int x = 0;
+        @Override
+        public void getVerificationCredentials() {
+            this.x += 1;
+        }
+
+        @Override
+        public void presentUserExistsMessage() {
+            this.x += 3;
+        }
+
+        @Override
+        public void getCode(int code) {
+            this.x += 5;
+        }
+
+        @Override
+        public void getUserCredentials(String username, String password, String email) {
+            this.x += 7;
+        }
+    }
+    TestOutputBoundary userExistsOutputBoundary = new TestOutputBoundary();
+    //The verification streams, and verification Stream factory used for testing
+    ISendVerificationCode verStream1 = (email, code) -> {
+
+    };
+    ISendVerificationCode verStream2 = (email, code) -> {
+
+    };
+    private class TestCreateMailMan implements createMailMan{
+        public int x;
+        @Override
+        public ISendVerificationCode getVerificationMethod(String type) {
+            if(type.equals("0")){
+                x = 100;
+                return verStream1;
+            }else{
+                x = 200;
+                return verStream2;
+            }
+        }
+    }
+    TestCreateMailMan mailManFactory = new TestCreateMailMan();
+
+    //In the first test, the user exists in the database, and we will set verStream1 as the verification method
+    @Test
+    public void userExistsInDatabase1(){
+        UserExistsPresenter uInteractor = new UserExistsPresenter(userExists,
```

Comment:
Although the class name UserExistsPresenter has presenter in it, would it be more of an interactor (as it seems to do core logic)?

Reply:
I asked the professor, and her opinion was that this is more of a presenter class, because it gives info to the use cases, so that it can perform its duties